dashboard/lib/dynamic_config/adapters/dynamodb_adapter.rb

```diff
@@ -0,0 +1,51 @@
+require 'aws-sdk'
+
+class DynamoDBAdapter
+  def initialize(table_name)
+    @table_name = table_name
+    @client = Aws::DynamoDB::Client.new
+  end
+
+  def get(key)
+    resp = @client.get_item({
+      table_name: @table_name,
+      key: {
+        'data-key' => JSON.dump(key)
+      }
+    })
+    return nil if resp.item.nil?
+    JSON.load(resp.item['data-value'])
+  end
+
+  def set(key, value)
+    @client.put_item({
+      table_name: @table_name,
+      item: {
+        'data-key' => JSON.dump(key),
```

Comment:
What is the expected behavior if the input is not JSON serializable?

JSON.dump unfortunately does not generate an error in this case. but rather generates invalid JSON that can't be parsed.  This could be a source of errors.
irb(main):006:0> x = Set.new([1,2,3])
=> #<Set: {1, 2, 3}>
irb(main):007:0> JSON.dump(x)
=> "\"#<Set:0x007fc6f4c7aae8>\""